dask/dataframe/tests/test_multi.py

```diff
@@ -796,6 +796,23 @@ def test_concat_with_operation_remains_hlg():
     assert_eq(result, expected)
 
 
+def test_concat_dataframe_empty():
+    df = pd.DataFrame({"a": [100, 200, 300]}, dtype="int64")
+    empty_df = pd.DataFrame([], dtype="int64")
+    df_concat = pd.concat([df, empty_df])
+
+    ddf = dd.from_pandas(df, npartitions=1)
+    empty_ddf = dd.from_pandas(empty_df, npartitions=1)
+    ddf_concat = dd.concat([ddf, empty_ddf])
+    assert_eq(df_concat, ddf_concat)
+
+    empty_df_with_col = pd.DataFrame([], columns=["x"], dtype="int64")
+    df_concat_with_col = pd.concat([df, empty_df_with_col])
+    empty_ddf_with_col = dd.from_pandas(empty_df_with_col, npartitions=1)
+    ddf_concat_with_col = dd.concat([ddf, empty_ddf_with_col])
+    assert_eq(df_concat_with_col, ddf_concat_with_col)
```

Comment:
This is currently failing, still figuring out why :/

Reply:
I've been trying to figure out what's going on here, and it seems that when we create the empty dataframe with a column name, `ddf._meta.dtype` in https://github.com/dask/dask/blob/78452267da9683fe637148fd7a667724c3ff7713/dask/dataframe/utils.py#L648-L651

returns dtype('O') but I can't seem to find how this gets assigned. 

One solution would be to compare the dtypes without using assert_eq, but I'm not sure that's the path we want to go. 

@jsignell do you have any suggestions on how to proceed here?

Reply:
This is certainly a weird one. If you inspect the objects created by the test you see that it's the index that is causing this issue. In particular for the case of `empty_df_with_col` you'll see that `empty_df_with_col.index.dtype == dtype('O')` that is just on the pandas object. On the other hand `df.index.dtype == dtype('int64')`. 

It seems odd to me that pandas would use a different default index dtype based on whether there are rows or not. But keeping going with the test, when `df` and `empty_df_with_col` are concatted, the output has `index.dtype == dtype('int64')`. In dask it does does not. Instead it has `index.dtype == dtype('O')`. 

I don't think there is actually a good way to handle this since the `_meta` can't depend on the len of the index. So I think you should just explicitly set the dtype of the index on `_meta` before doing `assert_eq`

Reply:
Thank you @ncclementi and @jsignell!